rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageExamplesTests.scala

```diff
@@ -0,0 +1,319 @@
+                 CityMatch(city = "Crystal Lake")
+               ),
+               new EntriesCaptor,
+               persist = false
+             )
+      _  = r1 shouldBe None
+      r2 <- space.produce(Channel("friends"), bob, persist = false)
+      _  = r2 shouldBe None
+      r3 <- space.produce(Channel("friends"), bob, persist = false)
+      _  = r3 shouldBe None
+      r4 <- space.produce(Channel("colleagues"), alice, persist = false)
+      _  = r4 shouldBe defined
+      _  = runK(r4)
+      _  = getK(r4).results shouldBe List(List(alice, bob, bob))
+    } yield (store.changes().map(_.isEmpty shouldBe true))
+
+  }
+
+  "CORE-365: A joined consume on multiple duplicate channels followed by the requisite produces" should
+    "return a continuation and the produced data" in withTestSpace { (store, space) =>
+    for {
+      r1 <- space
+             .consume(
+               List(
+                 Channel("family"),
+                 Channel("family"),
+                 Channel("family"),
+                 Channel("family"),
+                 Channel("colleagues"),
+                 Channel("colleagues"),
+                 Channel("colleagues"),
+                 Channel("friends"),
+                 Channel("friends")
+               ),
+               List(
+                 CityMatch(city = "Herbert"),
+                 CityMatch(city = "Herbert"),
+                 CityMatch(city = "Herbert"),
+                 CityMatch(city = "Herbert"),
+                 CityMatch(city = "Crystal Lake"),
+                 CityMatch(city = "Crystal Lake"),
+                 CityMatch(city = "Crystal Lake"),
+                 CityMatch(city = "Crystal Lake"),
+                 CityMatch(city = "Crystal Lake")
+               ),
+               new EntriesCaptor,
+               persist = false
+             )
+      _   = r1 shouldBe None
+      r2  <- space.produce(Channel("friends"), bob, persist = false)
+      r3  <- space.produce(Channel("family"), carol, persist = false)
+      r4  <- space.produce(Channel("colleagues"), alice, persist = false)
+      r5  <- space.produce(Channel("friends"), bob, persist = false)
+      r6  <- space.produce(Channel("family"), carol, persist = false)
+      r7  <- space.produce(Channel("colleagues"), alice, persist = false)
+      r8  <- space.produce(Channel("colleagues"), alice, persist = false)
+      r9  <- space.produce(Channel("family"), carol, persist = false)
+      r10 <- space.produce(Channel("family"), carol, persist = false)
+
+      _ = r2 shouldBe None
+      _ = r3 shouldBe None
+      _ = r4 shouldBe None
+      _ = r5 shouldBe None
+      _ = r6 shouldBe None
+      _ = r7 shouldBe None
+      _ = r8 shouldBe None
+      _ = r9 shouldBe None
+      _ = r10 shouldBe defined
+
+      _ = runK(r10)
+      _ = getK(r10).results shouldBe List(
+        List(carol, carol, carol, carol, alice, alice, alice, bob, bob)
+      )
+
+    } yield (store.changes().map(_.isEmpty shouldBe true))
+  }
```

Comment:
I'm feeling a bit lost in so many names.

Can you make this test use fewer channels in order to make the test easier to understand?

Reply:
This isn't the goal of the PR. The aim here is to adapt the existing https://github.com/rchain/rchain/blob/dev/rspace/src/test/scala/coop/rchain/rspace/StorageExamplesTests.scala to work with the recently introduced RSpace which uses `HotStore` under the hood. 

This means that any changes which aren't absolutely necessary for the tests to compile and pass aren't welcome. Simplifying them could be done in a different PR at a later stage